### PR TITLE
Fix scene.global.skybox is undefined if there is not SceneGlobals data in .scene file

### PR DESCRIPTION
**Pull request by dumganhar**
If JSB object is created from CPP, its `_ctor` method should be invoked too.

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/bindings/manual/jsb_conversions.h [hunk 1]
```diff
@@ -335,6 +335,12 @@ native_ptr_to_seval(T &v_ref, se::Value *ret, bool *isReturnCachedValue = nullpt
             se::Object *obj = se::Object::createObjectWithClass(cls);
             ret->setObject(obj, true);
             cc_tmp_set_private_data(obj, v);
+
+            se::Value property;
+            bool found = false;
+            found = obj->getProperty("_ctor", &property);
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_conversions.h]: Why not just write like this
```c++
bool found = obj->getProperty("_ctor", &property);
```

File: native/cocos/bindings/manual/jsb_conversions.h [hunk 2]
```diff
@@ -370,6 +376,12 @@ bool native_ptr_to_rooted_seval( // NOLINT(readability-identifier-naming)
             auto *obj = se::Object::createObjectWithClass(cls);
             obj->root();
             obj->setRawPrivateData(v);
+
+            se::Value property;
+            bool found = false;
+            found = obj->getProperty("_ctor", &property);
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_conversions.h]: The same as above.

File: native/cocos/bindings/manual/jsb_conversions.h (hunk 1 above)
Comment (minggo) [reply native/cocos/bindings/manual/jsb_conversions.h]: And may be just 
```c++
if (obj->getProperty("_ctor", &property)) property.toObject()->call(se::EmptyValueArray, obj)
```

File: native/cocos/bindings/manual/jsb_conversions.h [hunk 3]
```diff
@@ -424,27 +442,14 @@ bool native_ptr_to_seval(T *vp, se::Value *ret, bool *isReturnCachedValue = null
...
-            cc_tmp_set_private_data(obj, v);
-            if (isReturnCachedValue != nullptr) {
-                *isReturnCachedValue = false;
-            }
-        });
-
+    if (cls != nullptr) {
+        return native_ptr_to_seval<T>(vp, cls, ret, isReturnCachedValue);
+    } else {
+        ret->setNull();
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_conversions.h]: Need `  CC_ASSERT_NOT_NULL(cls); ` here

File: native/cocos/bindings/manual/jsb_conversions.h (hunk 1 above)
Comment (dumganhar) [reply native/cocos/bindings/manual/jsb_conversions.h]: DONE

File: native/cocos/bindings/manual/jsb_conversions.h (hunk 3 above)
Comment (dumganhar) [reply native/cocos/bindings/manual/jsb_conversions.h]: DONE

**Review by PatriceJiang — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/manual/jsb_conversions.h (hunk 1 above)
Comment (dumganhar) [reply native/cocos/bindings/manual/jsb_conversions.h]: OK, updated.

**Review by minggo — APPROVED** [review]

**Comment by dumganhar** [comment]
Don't merge now, I found an issue, I need to figure out whether it is caused by this PR.

**Pull request merged**